cli/generate_permissions.py

```diff
@@ -28,48 +28,77 @@
     from enum import Enum
     from typing import Dict, List
 
+    from ..shared.exceptions import PermissionException
     from .get_permission_parts import get_permission_parts
 
 
-    class OrganisationManagementLevel(str, Enum):
+    class CompareRightLevel(str):
+        def get_number(self, value) -> int:  # type: ignore
+            return 0
+
+        def __lt__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number < other_number
+
+        def __le__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number <= other_number
+
+        def __gt__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number > other_number
+
+        def __ge__(self, other: str) -> bool:
+            self_number = self.get_number(self)
+            other_number = self.get_number(other)
+            return self_number >= other_number
+
+
+    class OrganisationManagementLevel(CompareRightLevel, Enum):
         SUPERADMIN = "superadmin"
         CAN_MANAGE_USERS = "can_manage_users"
         CAN_MANAGE_ORGANISATION = "can_manage_organisation"
-
-        def __init__(self, oml: str):
-            super().__init__()
-            self.numbers = {
+        NO_RIGHT = "no_right"
```

Comment:
Mmh, I don't like this whole `no_right` thing, but I guess we'll leave it in for now, since I currently don't have a better idea, either.

Reply:
I'm with you;-)